pip/flatpak-pip-generator

```diff
@@ -74,15 +74,36 @@ def get_file_hash(filename: str) -> str:
             sha.update(data)
         return sha.hexdigest()
 
+
+def parse_continuation_lines(fin):
+    for line in fin:
+        line = line.rstrip('\n')
+        while line.endswith('\\'):
+            try:
+                line = line[:-1] + next(fin).rstrip('\n')
+            except StopIteration:
+                exit('Requirements have a wrong number of line continuation characters "\\"')
+        yield line
+
+
 packages = []
 if opts.requirements_file and os.path.exists(opts.requirements_file):
     with open(opts.requirements_file, 'r') as req_file:
-        packages = list(requirements.parse(req_file))
+        reqs = parse_continuation_lines(req_file)
+        reqs_as_str = '\n'.join([r.split('--hash')[0] for r in reqs])
+        packages = list(requirements.parse(reqs_as_str))
+        requirements_file = opts.requirements_file
 elif opts.packages:
     packages = list(requirements.parse('\n'.join(opts.packages)))
+    with tempfile.NamedTemporaryFile('w', delete=False) as req_file:
```

Comment:
This does mean you leave temp files around forever which isn't ideal.

Reply:
Oh at the end you had a deletion check. You just need to store this object around and let it get deleted.

Reply:
Does this commit resolve the issue?